src/main/java/org/isf/config/SecurityConfig.java

```diff
@@ -88,262 +87,253 @@ public PasswordEncoder encoder() {
 		return new BCryptPasswordEncoder();
 	}
 
+	@Value("${cors.allowed.origins}")
+	private String allowedOrigins;
+
 	@Bean
-	public CorsFilter corsFilter() {
-		UrlBasedCorsConfigurationSource source = new UrlBasedCorsConfigurationSource();
+	public CorsConfigurationSource corsConfigurationSource() {
+		CorsConfiguration configuration = new CorsConfiguration();
+		// Split the allowed origins string and add them to the configuration
+		for (String origin : allowedOrigins.split(",")) {
+			configuration.addAllowedOriginPattern(origin.trim());
+		}
+		configuration.setAllowedMethods(Arrays.asList("*")); // Allows all HTTP methods
+		configuration.addAllowedHeader("*"); // Allows all headers
+		configuration.setAllowCredentials(true); // Allows credentials (e.g., cookies)
+		configuration.setMaxAge(3600L); // Cache preflight responses for 1 hour
 
-		CorsConfiguration config = new CorsConfiguration();
-		config.addAllowedHeader("*");
-		// config.setAllowedHeaders(Arrays.asList("Accept", "Accept-Encoding", "Accept-Language", "Authorization", "Content-Type", "Cache-Control",
-		// "Connection", "Cookie", "Host", "Pragma", "Referer, User-Agent"));
-		config.setAllowedMethods(List.of("*"));
-		// config.setAllowedMethods(Arrays.asList("GET", "POST", "PUT", "DELETE"));
-		config.setAllowCredentials(true);
-		config.addAllowedOriginPattern("*");
-		config.setMaxAge(3600L);
-		source.registerCorsConfiguration("/**", config);
-		return new CorsFilter(source);
+		UrlBasedCorsConfigurationSource source = new UrlBasedCorsConfigurationSource();
+		source.registerCorsConfiguration("/**", configuration); // Apply configuration to all paths
+		return source;
 	}
 
 	@Bean
 	public SecurityFilterChain filterChain(HttpSecurity http) throws Exception {
-		http.sessionManagement().sessionCreationPolicy(SessionCreationPolicy.STATELESS)
-			.and().cors()
-			.and().csrf().disable().authorizeRequests()
-			// .expressionHandler(webExpressionHandler())
-			.and().exceptionHandling()
-			// .accessDeniedHandler(accessDeniedHandler)
-			.authenticationEntryPoint(restAuthenticationEntryPoint)
-			.and().authorizeRequests().requestMatchers("/auth/**").permitAll()
+		http
+						.cors(cors -> cors.configurationSource(corsConfigurationSource()))
```

Comment:
Indentation doesn't look correct. This is more than 4 spaces

Reply:
see https://github.com/informatici/openhospital-api/pull/478#issuecomment-2404734077
I'm trying to fix...

Reply:
Done! see https://github.com/informatici/openhospital-api/pull/478#issuecomment-2406141183